documentation/src/docs/wire-protocol.md

```diff
@@ -11,102 +11,97 @@ You may obtain a copy of the License at
 
-### EndAppendBlock- Request
+1.  `ConnectionId`: UUID (16 bytes).
+2.  `ConnectionOffset`: Long (8 bytes) (Data written so far over this connection to this Stream Segment).
+3.  Length of data before `EndAppendBlock` message - Integer (4 bytes). 
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffset - Long (8 bytes) Data written so far over this connection
-3.  Block Length - (4 Bytes) Total size of the block that was written. (Note this may more or less than the number of bytes between the BeginAppendBlock and this message)
+## EndAppendBlock- Request
 
-### Event - Request
+1.  `ConnectionId`: UUID (16 bytes).
+2.  `ConnectionOffset`: Long (8 bytes) Data written so far over this connection.
+3.  Block Length (4 bytes): Total size of the block that was written. (Note that this may be more or less than the number of bytes between the `BeginAppendBlock` and this message).
 
-Only valid inside of a block
+## Event - Request
+
+Only valid inside the block.
 
 1.  Data
 
-### Data Appended - Reply
+## Data Appended - Reply
 
-1.  ConnectionId - UUID (16 bytes)
-2.  ConnectionOffsetAckLevel - Long (8 bytes) The highest offset before which all data is successfully stored on this segment for this connectionId
+1.  `ConnectionId`: UUID (16 bytes).
+2.  `ConnectionOffsetAckLevel`: Long (8 bytes). The highest offset before which all the data is successfully stored on this Stream Segment for this `connectionId`.
 
-When appending a client
+When appending a client:
 
-1.  Establishes a connection to what it thinks is the correct host.
-2.  Sends a Setup Append request.
-3.  Waits for the Append Setup reply.
+- Establishes a connection to host chosen by it.
+- Sends a `Setup Append` request.
+- Waits for the `Append Setup reply`.
 
-Then it can
-1.  Send a BeginEventBlock request
-2.  Send as many messages as can fit in the block
-3.  Send an EndEventBlock request
+After receiving the `Append Setup` reply, it performs the following:
+- Send a `BeginEventBlock` request.
+- Send as many messages that can fit in the block.
+- Send an `EndEventBlock` request.
 
-While this is happening the server will be periodically sending it DataAppended replies acking messages. Note that there can be multiple "Appends" setup for a given TCP connection. This allows a client to share a connection when producing to multiple segments.
+While this is happening, the server will be periodically sending it `DataAppended` replies acking messages. Note that there can be multiple `Appends Setup` for a given TCP connection. This allows a client to share a connection when producing to multiple Segments.
 
-A client can optimize its appending by specifying a large value in it's BeginAppendBlock message, as the events inside of the block do not need to be processed individually.
+A client can optimize its appending by specifying a large value in it's `BeginAppendBlock` message, as the events inside of the block do not need to be processed individually.
```

Comment:
@tkaitchuck @howdy `A client can optimize its appending by specifying a large value in it's `BeginAppendBlock` message, as the events inside of the block do not need to be processed individually.`
need clarity??

Reply:
Events in the block are not processed individually. The whole block is appended without parsing it. So larger blocks are more efficient.

Reply:
Good! Accepted suggestion